ibm/service/power/data_source_ibm_pi_keys.go

```diff
@@ -13,78 +13,71 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
 
 	st "github.com/IBM-Cloud/power-go-client/clients/instance"
-	"github.com/IBM-Cloud/power-go-client/helpers"
 	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
 )
 
 func DataSourceIBMPIKeys() *schema.Resource {
 	return &schema.Resource{
 		ReadContext: dataSourceIBMPIKeysRead,
 		Schema: map[string]*schema.Schema{
-
 			// Arguments
 			Arg_CloudInstanceID: {
-				Type:         schema.TypeString,
+				Description:  "The GUID of the service instance associated with an account.",
 				Required:     true,
-				Description:  "PI cloud instance ID",
+				Type:         schema.TypeString,
 				ValidateFunc: validation.NoZeroValues,
 			},
 
 			// Attributes
 			Attr_Keys: {
-				Type:        schema.TypeList,
 				Computed:    true,
-				Description: "SSH Keys",
+				Description: "List of all the SSH keys.",
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
-						Attr_KeyName: {
-							Type:        schema.TypeString,
+						Attr_KeyCreationDate: {
```

Comment:
Let's use either of `Attr_KeyCreationDate` or `Attr_CreationDate` in both key and keys data sources.

Reply:
Made changes to Attr_CreationDate